client/src/components/DisplayTable/DisplayTableUtils.js

```diff
@@ -62,3 +63,229 @@ export const DisplayTableDownloadCsv = ({ csv_string, table_name }) => (
     )}
   </Fragment>
 );
+
+const _is_ellipsis = (page_options, page_num, index) =>
+  index !== 0 &&
+  index !== page_options.length - 1 &&
+  (page_options[index + 1] !== page_num + 1 ||
+    page_options[index - 1] !== page_num - 1);
+
+export const SelectPage = ({
```

Comment:
To be honest, I'd want to try and clean this up a lot more, just so future devs will have an easier time maintaining it... but I've let this PR get held up too long already (sorry about that) and at least this is relatively isolated code. Not likely to need maintenance unless someone wants to change the behaviour, in which case they can just rewrite it rather than try and modify it, haha.

Maybe just to indicate to future devs that they don't need to try too hard to understand it unless they really need to, add a comment like `TODO cleanup or refactor`. I'd be fine with merging it in after that.